zio-http/src/main/scala/zio/http/Server.scala

```diff
@@ -30,11 +30,22 @@ object Server {
     ZIO.serviceWithZIO[Server](_.install(httpApp, errorCallback)) *> ZIO.service[Server].map(_.port)
   }
 
+  def defaultWithPort(port: Int): ZLayer[Any, Throwable, Server] =
+    defaultWith(_.port(port))
+
+  def defaultWith(f: ServerConfig => ServerConfig): ZLayer[Any, Throwable, Server] =
+    live(f(ServerConfig.default))
+
   val default: ZLayer[Any, Throwable, Server] = {
     implicit val trace = Trace.empty
     ServerConfig.live >>> live
   }
 
+  def live(conf: ServerConfig): ZLayer[Any, Throwable, Server] = {
```

Comment:
If this is already going to be a `def` we can just accept a `Trace` as a parameter.